makeWorkspace/make_ws.py

```diff
@@ -41,90 +150,13 @@ def main():
     os.makedirs(outdir)
 
   fin = ROOT.TFile(args.file,'READ')
-  f_jes = ROOT.TFile("sys/shape_jes_uncs_smooth.root")
   fout = ROOT.TFile(args.out,'RECREATE')
   dummy = []
   for category in args.categories:
-    fdir = fin.Get("category_"+category)
-
-    foutdir = fout.mkdir("category_"+category)
-
-    wsin_combine = ROOT.RooWorkspace("wspace_"+category,"wspace_"+category)
-    wsin_combine._import = SafeWorkspaceImporter(wsin_combine)#getattr(wsin_combine,"import")
-
-    variable_name = "mjj" if ("vbf" in category) else "met"
-    varl = ROOT.RooRealVar(variable_name, variable_name, 0,100000);
-    # Helper function
-    def write_obj(obj, name):
-      '''Converts histogram to RooDataHist and writes to workspace + ROOT file'''
-      print "Creating Data Hist for ", name
-      dhist = ROOT.RooDataHist(
-                            name,
-                            "DataSet - %s, %s"%(category,name),
-                            ROOT.RooArgList(varl),
-                            obj
-                            )
-      wsin_combine._import(dhist)
-
-      # Write the individual histograms
-      # for easy transfer factor calculation
-      # later on
-      obj.SetDirectory(0)
-      foutdir.cd()
-      foutdir.WriteTObject(obj)
-
-    # Loop over all keys in the input file
-    # pick out the histograms, and add to
-    # the work space.
-    keys_local = fdir.GetListOfKeys()
-    for key in keys_local:
-      obj = key.ReadObj()
-      if type(obj) not in [ROOT.TH1D, ROOT.TH1F]:
-        continue
-      title = obj.GetTitle()
-      name = obj.GetName()
-
-      # Ensure non-zero integral for combine
-      if not obj.Integral() > 0:
-        obj.SetBinContent(1,0.0001)
-
-      # Add overflow to last bin
-      overflow        = obj.GetBinContent(obj.GetNbinsX()+1)
-      overflow_err    = obj.GetBinError(obj.GetNbinsX()+1)
-      lastbin         = obj.GetBinContent(obj.GetNbinsX())
-      lastbin_err     = obj.GetBinError(obj.GetNbinsX())
-      new_lastbin     = overflow + lastbin
-      new_lastbin_err = sqrt(overflow_err**2 + lastbin_err**2)
-
-      obj.SetBinContent(obj.GetNbinsX(), new_lastbin)
-      obj.SetBinError(obj.GetNbinsX(), new_lastbin_err)
-      obj.SetBinContent(obj.GetNbinsX()+1, 0)
-      obj.SetBinError(obj.GetNbinsX()+1, 0)
-
-      write_obj(obj, name)
-      
-      # Systematic variations
-      if not 'data' in name:
-        channel = re.sub("(loose|tight)","", category)
-        for key in [x.GetName() for x in f_jes.GetListOfKeys()]:
-          if not (channel in key):
-            continue
-          variation = key.replace(channel+"_","")
-          name = obj.GetName()+"_"+variation
-          varied_obj = obj.Clone(name)
-          varied_obj.Multiply(f_jes.Get(key))
-          write_obj(varied_obj, name)
-
-
+    f_jes = get_jes_file(category)
+    wsin_combine = create_workspace(fin, f_jes, fout, category)
```

Comment:
Ok, just one tiny thing: Can we just move the call to ```get_jes_file``` into ```create_workspace```? Since we do not use f_jes anywhere else here, we can reduce the number of rgs to `create_workspace`

Reply:
Sure sounds good, I can move the function call get_jes_file into create_workspace